### ISSUE TITLE
About Wordbook

### ISSUE THREAD

**Issue by fastrizwaan** [fushinari/Wordbook#15]
I found out about wordbook from https://www.reddit.com/r/flatpak/comments/kifmdc/artha_dictionary_flatpak/
![Screenshot from 2020-12-26 01-09-17](https://user-images.githubusercontent.com/1161682/103141204-3db4a180-4717-11eb-9f9a-932364b9048a.png)

![ss2](https://user-images.githubusercontent.com/1161682/103141208-4d33ea80-4717-11eb-9dfb-bc889cb521a3.png)

I like the coloring used by you. Artha is gtk2 application.

I was making flatpak for Artha, and successfully made it.
https://github.com/fastrizwaan/Artha_flatpak

**Comment by fastrizwaan** [comment]
You know, we should have global shortcut support for launching wordbook, to show results for selections in other windows. Artha does that with gsettings/dconf.

**Comment by fushinari** [comment]
Artha being outdated was the reason I made Wordbook. I originally tried to port Artha to GTK in fact and found out that it was just not worth the effort.

You can set a global shortcut for the app itself and then do Ctrl+Shift+S and it will take text from selection clipboard, which is whatever text you last selected. I might simplify this later.

If you want to test Wordbook as it is right now in `master`, you can remove the current `python-wn` package you have and install the one from [nltk/wordnet](https://github.com/nltk/wordnet).

**Comment by fastrizwaan** [comment]
Just installed nltk/wordnet. It's fast, it's featureful, looks good.
Here's the install instructions for others
```
git clone https://github.com/nltk/wordnet.git
cd wordnet/
sudo python setup.py install
cd ..
git clone https://github.com/fushinari/wordbook.git
cd wordbook
sudo python setup.py install
wordbook
```
![Screenshot from 2020-12-26 02-27-04](https://user-images.githubusercontent.com/1161682/103141930-e962ef00-4721-11eb-8d26-630e679b8557.png)

I miss these features:
1. History (esp. if we want to see the last random word)
2. wordbook.desktop (not visible in gnome shell)
3. wordbook icon (https://www.google.com/search?q=book+icon&tbm=isch&ved=2ahUKEwjz19-cg-rtAhXKFHIKHZQ8BdwQ2-cCegQIABAA&oq=book+icon&gs_lcp=CgNpbWcQAzIFCAAQsQMyBQgAELEDMgIIADICCAAyAggAMgIIADICCAAyAggAMgIIADICCAA6BAgjECc6BAgAEENQpNYKWPrdCmDJ3wpoAHAAeACAAcUBiAHwCZIBAzAuOZgBAKABAaoBC2d3cy13aXotaW1nwAEB&sclient=img&ei=tlLmX_OkGcqpyAOU-ZTgDQ&bih=659&biw=856)
4. system shortcut like artha or goldendict
5. release on flathub

I'm impressed, where did you learn pygtk and libhandy? I would also like to learn, I made a rudimentary vocabulary builder software using pygtk3. https://github.com/fastrizwaan/VocabularyBuilder

**Comment by fushinari** [comment]
2, 3 and 5 will be solved soon. 1 a bit later because I'm waiting for HdyFlap to land in a libhandy release. 4 I'm not really concerned about but may facilitate (not implement in-app, just make it easier for users to do) later.

The best resource has been other people's code. Especially the code of applications like GNOME Music, Lollypop, Komikku, etc have been really helpful. Working on Dialect has also been a great experience and taught me a lot.

**Comment by fushinari** [comment]
I think all questions here have been answered. I'll address the feature requests asap as I've explained above. Thanks for trying it out, getting a third-party opinion on these matters has been helpful. :)

(2, 3  and 5 have already been at least partially addressed in #13 btw.)
